internal/provider/network/isolated_resource.go

```diff
@@ -0,0 +1,670 @@
+			return
+		}
+	} else {
+		// OwnerID is not null, so we need to retrieve the URN via Owner_ID (VDC or VDC Group)
+		// First, we need to check if it is a VDC
+		vdc, err = org.GetVDCById(plan.OwnerID.ValueString(), true)
+		if err == nil {
+			plan.OwnerID = types.StringValue(vdc.Vdc.ID)
+		} else { // It is not a VDC, so it may be a VDC Group
+			urn, err = govcd.BuildUrnWithUuid("urn:vcloud:vdcGroup:", plan.OwnerID.ValueString())
+			if err == nil {
+				vdcGroup, err = org.GetVdcGroupById(urn)
+				if err != nil {
+					resp.Diagnostics.AddError("[CREATE] Error retrieving VDC Group from OwnerID", err.Error())
+					return
+				}
+
+				// OwnerId is VDCGroup ID
+				isGroup = true
+				plan.OwnerID = types.StringValue(urn)
+			} else {
+				resp.Diagnostics.AddError("[CREATE] Error retrieving VDC or VDC Group from OwnerID", err.Error())
+				return
+			}
+		}
+	}
+
+	// Lock VDC or VDC Group to prevent concurrent access
+	vcdMutexKV := mutex.NewKV()
+	vcdMutexKV.KvLock(ctx, urn)
+	defer vcdMutexKV.KvUnlock(ctx, urn)
+
+	// Get network type
+	ipPool := []staticIPPoolResourceModel{}
+	resp.Diagnostics.Append(plan.StaticIPPool.ElementsAs(ctx, &ipPool, true)...)
+	myshared := false // Cloudavenue does not support shared networks
+	networkType := &govdctypes.OpenApiOrgVdcNetwork{
+		Name:        plan.Name.ValueString(),
+		Description: plan.Description.ValueString(),
+		Shared:      &myshared,
+		NetworkType: govdctypes.OrgVdcNetworkTypeIsolated,
+		Subnets: govdctypes.OrgVdcNetworkSubnets{
+			Values: []govdctypes.OrgVdcNetworkSubnetValues{
+				{
+					Gateway:      plan.Gateway.ValueString(),
+					PrefixLength: int(plan.PrefixLength.ValueInt64()),
+					IPRanges: govdctypes.OrgVdcNetworkSubnetIPRanges{
+						Values: myProcessIPRanges(ipPool),
+					},
+					DNSServer1: plan.PrimaryDNS.ValueString(),
+					DNSServer2: plan.SecondaryDNS.ValueString(),
+					DNSSuffix:  plan.SuffixDNS.ValueString(),
+				},
+			},
+		},
+	}
+
+	// Set OwnerRef if VDC Group
+	if isGroup {
+		networkType.OwnerRef = &govdctypes.OpenApiReference{ID: vdcGroup.VdcGroup.Id}
+	} else {
+		networkType.OwnerRef = &govdctypes.OpenApiReference{ID: vdc.Vdc.ID}
+	}
+
+	// Create network
+	orgNetwork, err := org.CreateOpenApiOrgVdcNetwork(networkType)
+	if err != nil {
+		resp.Diagnostics.AddError("[CREATE] Error creating isolated network", err.Error())
+		return
+	}
+
+	// set Plan
+	plan = &networkIsolatedResourceModel{
+		ID:            types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID),
+		OwnerID:       types.StringValue(extractUUID(plan.OwnerID.ValueString())),
+		Name:          plan.Name,
+		Description:   plan.Description,
+		Gateway:       plan.Gateway,
+		PrefixLength:  plan.PrefixLength,
+		StaticIPPool:  plan.StaticIPPool,
+		PrimaryDNS:    plan.PrimaryDNS,
+		SecondaryDNS:  plan.SecondaryDNS,
+		SuffixDNS:     plan.SuffixDNS,
+		MetadataEntry: plan.MetadataEntry,
+	}
```

Comment:
Why using `extractUUID` if this field is Only UUID (Description and Validator)


```suggestion
	// set Plan
    plan.ID = types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID)
    plan.OwnerID = types.StringValue(extractUUID(plan.OwnerID.ValueString()))
```